src/foam/core/XMLSupport.java

```diff
@@ -0,0 +1,158 @@
+                            // Create new fObject
+                            String objClass = xmlr.getAttributeValue(null, "class");
+                            Class<?> cls;
+
+                            try {
+                                cls = Class.forName(objClass);
+                                clsInstance = cls.newInstance();
+                                ClassInfo csInfo = ((FObject)clsInstance).getClassInfo();
+                                xmlDAO.setOf(csInfo);
+                            } catch (ClassNotFoundException | InstantiationException ex) {
+
+                            }
+                            // Object properties
+                            fillProperties((FObject) clsInstance, xmlr);
+                            if (clsInstance != null){
+                                xmlDAO.putOnly( (FObject) clsInstance);
+                            }
+                        }
+                        break;
+                    case XMLStreamConstants.END_ELEMENT:
+                        break;
+                    case XMLStreamConstants.END_DOCUMENT:
+                        break;
+                }
+            }
+            xmlr.close();
+        } catch (FileNotFoundException ex) {
+            throw new FileNotFoundException("Xml file not found with given filename: " + fileName);
+        } catch (XMLStreamException ex) {
+            throw new XMLStreamException("Premature end of xml file");
+        }
+    }
+
+    private static void fillProperties(FObject obj, XMLStreamReader reader) throws XMLStreamException {
+        try {
+            int eventType;
+            PropertyInfo prop = null;
+            while (reader.hasNext()) {
+                eventType = reader.next();
+                switch (eventType){
+                    case XMLStreamConstants.START_ELEMENT:
+                        prop = (PropertyInfo) obj.getClassInfo().getAxiomByName(reader.getLocalName());
+                        if (prop == null) {
+                            throw new XMLStreamException("Could not find property " + reader.getLocalName());
+                        }
+                        break;
+                    case XMLStreamConstants.CHARACTERS:
+                        if (prop != null) {
+                            prop.set(obj, reader.getText());
+                            prop = null;
+                        }
+                        break;
+                    case XMLStreamConstants.END_ELEMENT:
+                        if(reader.getLocalName().equals("object")) { return; }
+                        break;
+                }
+            }
+        } catch (XMLStreamException ex ) {
+            throw new XMLStreamException("Premature end of xml file");
+        }
+    }
+
+    // FObjects taken from XMLDAO and written to file with specified filename given
+    public static void FObjectToXML (String fileName, XMLDAO xmlDAO) throws XMLStreamException {
```

Comment:
Have second arg be an XMLStreamWriter. Add a second convenience method which takes a filename as second arg and then calls this one. Remove all references to XMLDAO.